RVR parsing should not ignore RVR strings without 'FT'.

wingunder opened the pull request:
The following RVR string is valid, but got skipped:
  R14/1300VP2000D
This commit fixes this issue.

wingunder committed:
RVR parsing should not ignore RVR strings without 'FT'.

The following RVR string is valid, but got skipped:
  R14/1300VP2000D
This commit fixes this issue.

bovine commented:
Was this format occurring regularly at some airports?

Are those airports reporting distances in meters instead of feet?

wingunder commented:
> Was this format occurring regularly at some airports?

Yes, basically everywhere in Europe, at least. Just search for any airport in northern Europe during a nice winter storm and you'll see that the RVR is in meters, not feet. Basically all horizontal distance measurements are in meters, in Europe, but note that the vertical distances are still in feet.

>Are those airports reporting distances in meters instead of feet?

Yes, but only for horizontal measurements, as stated above.

BTW, the print_decoded_metar.c source should also reflect meters instead of feet in this case. I wanted to change that as well, but I'll have to change at least 'struct runway_VisRange' for this. It then turned out that I had to modify two header files, with the same modification. This lead me to create another pull request. Once this is solved, I could start fixing this meters/feet problem in the c structs.

bovine commented:
ok, i merged pr #5 if you want to make further changes to this PR that depend on that

wingunder commented:
Thanks :+1:

wingunder committed:
Merge remote-tracking branch 'org/master'

wingunder committed:
Added a Distance_Unit type (FT/METERS) for RVR.

The default remains FT, but if the parser does not find the FT token,
it swithes to METERS for RVR.

At least North America uses feet, whereas at least Europe uses meters,
for RVR measurements.

wingunder committed:
Added an example with RVR in meters.

The FT token is not present after the RVR distance token.

bovine commented on include/metar.h:
```diff
@@ -42,7 +42,9 @@ typedef unsigned short int MDSP_BOOL;
 #define MAX_PARTIAL_OBSCURATIONS 2
- 
+
+typedef enum distance_unit { FT, METERS } Distance_Unit;
```
For consistency, maybe change `FT` to be `FEET` since you spell out METERS?

Also I wonder if it would be better to use longer names for these enums to avoid possibly name collisions with people's programs? Maybe something like DIST_FEET and DIST_METERS?

wingunder committed:
Improved the naming of Distance_Unit items.

This renaming was done, as namespace could be an issue in downstream
programs that use this library.

Thanks @bovine, for the suggestions.

wingunder commented:
Thanks @bovine, I changed it according to your suggestions.

bovine commented:
Another thought that occurred to me is that this is technically a compatibility-breaking change... Existing applications written against the current version of our library might be unexpectedly surprised if they upgrade to the new version of this library without ensuring they change their code to actually check the new `distance_unit` member. They may be surprised to start receiving wildly out-of-range values for airports that report in meters.

wingunder commented:
Thanks, I totally accept your point.

I guess either none of the downstream apps deal with non-FT METARs, or no-one has realized that RVR is silently ignored, for a non-FT METARs. The only 2 things that I can think of in this case are:

   * convert meters to feet and supply feet-only-RVR to downstream apps. This will however add another inconsistency, as I think it will be pretty strange if VIS get supplied in meters and RVR in feet, for the same decoded METAR. This would mean _everyting_ should be converted to feet, for consistency, which _could_ also potentially break downstream apps.
   * create a new interface for downstream apps that is FT/M friendly.

It's your call. I'm not really relying on this code. I just happen to have spotted the RVR mistake and tried to correct it.
